ls1intum/Artemis
# [Feature] Stash unsubmitted changes after due date has passed

### Checklist

### Motivation and Context


We want to display only the changes that the student submitted (all changes that are on BitBucket). To ensure this, we will stash all unsubmitted changes.

### Description

- Implemented method to  stash changes after the due date has passed
- Invoke stash changes method in a scheduler, that will be executed when the due date has passed (same as for lock repo)

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create Programming Exercise, set due date in the near future and enable manual assessment
4. Participate on exercise
4.1 Submit code and make change and just click on save (do not submit a second time!!!)
4.2 Just save your changes (do not submit at all)
4.3 Make changes and alway submit your code
5. Wait until the due date has passed
6. Check your submission in the code editor and check that the not saved changes are not displayed anymore

### Test Coverage

* GitService.java: 80%
* ProgrammingExerciseParticipationService,java: 86%

### Screenshots


During participation after clicking on save:
<img width="1665" alt="image" src="https://user-images.githubusercontent.com/41108487/97924909-b83cf280-1d60-11eb-82ef-1a4fa0ddd33a.png">

After due date:
<img width="1651" alt="image" src="https://user-images.githubusercontent.com/41108487/97925158-34cfd100-1d61-11eb-8a39-cf93edf87c15.png">
